vespaclient-container-plugin/src/main/java/com/yahoo/document/restapi/resource/DocumentV1ApiHandler.java

```diff
@@ -406,18 +409,14 @@ private void enqueueAndDispatch(HttpRequest request, ResponseHandler handler, Su
             return;
         }
         Operation operation = Operation.lazilyParsed(request, handler, operationParser);
-        if (enqueued.get() == 0 && operation.dispatch()) // Bypass queue if it is empty.
+        if (enqueued.get() == 1 && operation.dispatch()) // Bypass queue if it is empty.
```

Comment:
`enqueued` might be positive even when the actual queue is empty (when multiple threads executes `enqueueAndDispatch` simultaneously.